app/questionnaire/questionnaire_store_updater.py

```diff
@@ -1,5 +1,17 @@
 from typing import List, Tuple
+
 from app.data_model.answer_store import Answer
+from app.questionnaire.location import Location
+from app.questionnaire.routing_path import SectionRoutingPath
+
+
+def _is_last_block_on_path(
+    block_id: str, routing_path: SectionRoutingPath
+) -> bool:
+    try:
+        return block_id == routing_path[-1].block_id
```

Comment:
```suggestion
        return hasattr(routing_path[-1], 'block_id') and block_id == routing_path[-1].block_id
```

Using hasattr would save the need for the try here.